testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDatabricksColumnMappingMode.java

```diff
@@ -0,0 +1,175 @@
+import static io.trino.tempto.assertions.QueryAssert.assertQueryFailure;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.tests.product.TestGroups.DELTA_LAKE_DATABRICKS;
+import static io.trino.tests.product.TestGroups.DELTA_LAKE_OSS;
+import static io.trino.tests.product.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static io.trino.tests.product.hive.util.TemporaryHiveTable.randomTableSuffix;
+import static io.trino.tests.product.utils.QueryExecutors.onDelta;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+
+public class TestDeltaLakeDatabricksColumnMappingMode
+        extends BaseTestDeltaLakeS3Storage
+{
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS})
+    public void testColumnMappingModeName()
+    {
+        String tableName = "test_dl_column_mapping_mode_" + randomTableSuffix();
+
+        onDelta().executeQuery("" +
+                "CREATE TABLE default." + tableName +
+                " (a_number INT, array_col ARRAY<STRUCT<array_struct_element: STRING>>, nested STRUCT<field1: STRING>, a_string STRING)" +
+                " USING delta " +
+                " LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "'" +
+                " TBLPROPERTIES (" +
+                " 'delta.columnMapping.mode' = 'name'," +
+                " 'delta.minReaderVersion' = '2'," +
+                " 'delta.minWriterVersion' = '5')");
+
+        try {
+            onDelta().executeQuery("" +
+                    "INSERT INTO default." + tableName + " VALUES " +
+                    "(1, array(struct('nested 1')), struct('databricks 1'),'ala'), " +
+                    "(2, array(struct('nested 2')), struct('databricks 2'), 'kota')");
+
+            List<Row> expectedRows = ImmutableList.of(
+                    row(1, "nested 1", "databricks 1", "ala"),
+                    row(2, "nested 2", "databricks 2", "kota"));
+
+            assertThat(onDelta().executeQuery("SELECT a_number, array_col[0].array_struct_element, nested.field1, a_string FROM default." + tableName))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery("SELECT a_number, array_col[1].array_struct_element, nested.field1, a_string FROM delta.default." + tableName))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery("SELECT a_string FROM delta.default." + tableName + " WHERE a_number = 1"))
+                    .containsOnly(ImmutableList.of(row("ala")));
+            assertThat(onTrino().executeQuery("SELECT a_number FROM delta.default." + tableName + " WHERE nested.field1 = 'databricks 1'"))
+                    .containsOnly(ImmutableList.of(row(1)));
+
+            // Verify the connector can read renamed columns correctly
+            onDelta().executeQuery("ALTER TABLE default." + tableName + " RENAME COLUMN a_number TO new_a_column");
```

Comment:
Does Spark allow renaming struct column fields? `rename column nested.field1 to nested.field2` would be a good addition if it does

Reply:
Yes, Spark can rename struct column fields. Added another `RENAME COLUMN` statement.